source/features/index.tsx

```diff
@@ -104,12 +105,81 @@ const globalReady: Promise<RGHOptions> = new Promise(async resolve => {
 		document.head.append(<style>{options.customCSS}</style>);
 	}
 
+	const testedFeatures: [string[], string[]] | false | undefined = await cache.get('bisect');
+	if (testedFeatures !== undefined) {
+		if (testedFeatures === false) {
+			createMessageBox('Every feature has been disabled. If you still see the bug, try disabling the whole extension.', false);
+			return;
+		}
+
+		const [firstHalf] = testedFeatures;
+		options = {
+			...options,
+			...Object.fromEntries(__features__.map(feature => [`feature:${feature}`, firstHalf.includes(feature)]))
+		};
+
+		if (firstHalf.length === 0) {
+			createMessageBox('Every feature has been disabled. Can you see the bug?');
+			return;
+		}
+
+		if (firstHalf.length === 1) {
+			createMessageBox(<span>The bug is caused by <a href={'https://github.com/sindresorhus/refined-github/blob/main/source/features/' + firstHalf[0] + '.tsx'}><code>{firstHalf[0]}</code></a>.</span>, false);
+			return;
+		}
+
+		createMessageBox(`${Math.ceil(Math.log2(firstHalf.length))} steps remaining. Can you see the bug?`);
+	}
+
 	// Create logging function
 	log = options.logging ? console.log : () => {/* No logging */};
 
 	resolve(options);
 });
 
+function createMessageBox(message: string | Element, yesNoButtons = true): void {
+	delegate(document, '.rgh-bisect-button-choice', 'click', onBisectButtonChoiceClick);
+	delegate(document, '.rgh-bisect-button-end', 'click', onBisectButtonEndClick);
+
+	document.body.append(
+		<div className="Box" style={{position: 'fixed', top: 0, left: 0}}>
+			<p>{message}</p>
+			{yesNoButtons ? <button type="button" className="btn btn-danger rgh-bisect-button-choice" data-answer="no">No</button> : undefined}
+			{yesNoButtons ? <button type="button" className="btn btn-primary rgh-bisect-button-choice" data-answer="yes">Yes</button> : undefined}
+			{yesNoButtons ? undefined : <button type="button" className="btn btn-primary rgh-bisect-button-end">OK</button>}
+		</div>
+	);
+}
+
+async function onBisectButtonChoiceClick({target}: MouseEvent): Promise<void> {
+	const {answer} = (target as HTMLElement).dataset;
+	let testedFeatures: [string[], string[]] | false | undefined = await cache.get('bisect');
+	if (!testedFeatures) {
+		return;
+	}
+
+	const [firstHalf, secondHalf] = testedFeatures;
+	if (firstHalf.length === 0) {
+		testedFeatures = answer === 'yes' ? false : splitArray(__features__);
+	} else {
+		testedFeatures = splitArray(answer === 'yes' ? firstHalf : secondHalf);
+	}
+
+	await cache.set('bisect', testedFeatures);
+	await browser.runtime.sendMessage({reloadTab: true});
+}
+
+async function onBisectButtonEndClick(): Promise<void> {
+	await cache.delete('bisect');
+	await browser.runtime.sendMessage({reloadTab: true});
+}
+
+function splitArray<T>(array: T[]): [T[], T[]] {
+	const half = Math.ceil(array.length / 2);
+
+	return [array.slice(0, half), array.slice(half)];
+}
```

Comment:
Should this be moved to a helper module?

Reply:
Wouldn't be surprised if sindre has a package that does this

Reply:
He does. Chunk-array or something like that

Reply:
Instead of splitting the array, you just need to store the whole list you're testing at the moment, you'll automatically know that half of them are currently disabled.

For example, on the step where 50% of the features are disabled, you store 100% of them and enable only the first half.

Then on the next step, you store 50% of them and enable half of that.

Etc. No need to store split arrays if you already know they're always going to be split in half. We just need to ensure that the math is right and consistent when there's an odd number of features.

Reply:
> We just need to ensure that the math is right and consistent when there's an odd number of features

That's what the `Math.ceil()` is for: it ensures there's always at least one feature in the first half (so if it's empty, it means all features have been disabled).